lib/src/list_multimap/list_multimap_builder.dart

```diff
@@ -158,6 +156,10 @@ class ListMultimapBuilder<K, V> {
     _builderMap.clear();
   }
 
+  /// As [ListMultimap], but results are [ListBuilder]s.
+  ListBuilder<V> operator [](Object key) =>
+      key is K ? _getValuesBuilder(key) : new ListBuilder<V>();
```

Comment:
Same here; the `ListBuilder` may or may not be associated with anything. Could it be `BuiltList`?

Another option would be to make it `Iterable` and return an unmodifiable view of ListBuilder's internal list.

Reply:
In this case, returning the ListBuilder is important to allow this to be used for modifications, e.g., `b[key].insert(index, value)` to insert an element in a specific location. This is consistent with ListMultimap where "updates to the [returned] collection modify the multimap and likewise, modifications to the multimap are reflected in the returned collection."

This could be changed to only accept K, or to throw an exception, so the result is always associated with a key. Afaict, in ListMultimap the delegate map is silently recast to Map<Object, ...> and the key added so we can't be consistent with ListMultimap here.

Reply:
Hmm, so it looks like the returned `ListBuilder` will stay attached unless `removeAll` is called ... at which point it detaches.

We could fix that by modifying `removeAll` so it just empties the builder if it's there, instead of assigning a new one? At that point it seems like it's reasonably well defined.

Reply:
Good catch! Changed `removeAll` to clear the existing `ListBuilder` if there is one (I'll add a test as well). `clear` will still detach all existing `ListBuilder`s but it's not really clear whether that's the correct behaviour or not; do you have a preference?

Reply:
Thanks!

Hmm, `replace` will also have the effect of detaching builders.

Given that `replace` and `clear` are usually "destroy everything" type operations, I think it's okay that they detach existing builders. I guess, add a note to that effect to the `replace` and `clear` docs? Something along the lines of 'any [ListBuilder]s associated with this collection are disconnected'?

Reply:
SG, done.